src/openms/include/OpenMS/ANALYSIS/TOPDOWN/FLASHIda.h

```diff
@@ -0,0 +1,192 @@
+           @param mzs mz values of the input spectrum
+           @param intensities intensities of the input spectrum
+           @param length length of mzs and ints
+           @param rt Retention time in seconds
+           @param ms_level ms level
+           @param name spectrum name
+           @return number of peak groups
+      */
+    int getPeakGroups(const double *mzs,
+                      const double *intensities,
+                      const int length,
+                      const double rt,
+                      const int ms_level,
+                      const char *name);
+
+    /**
+           @brief get isolation windows using FLASHDeconv algorithm. Many parameters are in primitive types so they can be passed to C# FLASHIda side.
+           All parameters are for isolation windows.
+           @param window_start window start mzs
+           @param window_end window end mzs
+           @param qscores QScores of windows
+           @param charges charges of windows
+           @param min_charges minimum charges
+           @param max_charges maximum charges
+           @param mono_masses monoisotopic masses
+           @param charge_cos charge cosine scores
+           @param charge_snrs charge SNRs or precursor SNRs
+           @param iso_cos mass cosine scores
+           @param snrs mass SNRs
+           @param charge_scores charge distribution scores
+           @param ppm_errors average PPM errors
+           @param precursor_intensities precursor peak intensities
+           @param peakgroup_intensities precursor mass intensities
+      */
+    void getIsolationWindows(double *window_start,
+                             double *window_end,
+                             double *qscores,
+                             int *charges,
+                             int *min_charges,
+                             int *max_charges,
+                             double *mono_masses,
+                             double *charge_cos,
+                             double *charge_snrs,
+                             double *iso_cos,
+                             double *snrs, double *charge_scores,
+                             double *ppm_errors,
+                             double *precursor_intensities,
+                             double *peakgroup_intensities);
+
+  private:
+    /// PeakGroup comparator for soring by QScore
+    struct
+    {
+      bool operator()(const PeakGroup &a, const PeakGroup &b) const
+      {
+        return a.getQScore() > b.getQScore();
+      }
+    } QscoreComparator_;
+
+    /// Maps that are necessary for mass exclusion
+    std::unordered_map<int, double> mz_rt_map_;
+    std::unordered_map<int, double> mass_rt_map_;
+    std::unordered_map<int, double> all_mass_rt_map_;
+    std::unordered_map<int, double> mass_qscore_map_;
+
+    /**
+         @brief discard peak groups using mass exclusion
+         @param spec input spectrum
```

Comment:
spec is no parameter of this method - remove?

Reply:
Sure